pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/AggregationFunctionContext.java

```diff
@@ -18,19 +18,28 @@
  */
 package org.apache.pinot.core.query.aggregation;
 
+import com.google.common.base.Preconditions;
+import java.util.List;
 import org.apache.pinot.core.query.aggregation.function.AggregationFunction;
+import org.apache.pinot.core.query.aggregation.function.AggregationFunctionUtils;
 
 
 /**
  * This class caches miscellaneous data to perform efficient aggregation.
+ *
+ * TODO: Remove this class, as it no longer provides any value after aggregation functions now store
```

Comment:
+1

Reply:
why not remove it in this PR itself?

Reply:
Seemed like a bit non-trivial, will address in the next PR.